lib/web/websocket/connection.js

```diff
@@ -189,11 +189,6 @@ function establishWebSocketConnection (url, protocols, ws, onEstablish, options)
           failWebsocketConnection(ws, 'Protocol was not set in the opening handshake.')
           return
         }
-      } else if (request.headersList.contains('sec-websocket-protocol', true)) {
```

Comment:
I think this check should be left though. 

In spec:
> In this version of the protocol,
>    the main option field is |Sec-WebSocket-Protocol|, which indicates
>    the subprotocol that the server has selected.  WebSocket clients
>    verify that the server included one of the values that was specified
>    in the WebSocket client's handshake.

Logically, this is an exchange of information about capabilities between the client and server. And if the server doesn't support what the client wants, then the connection should be impossible.

I checked how it works in the browser. And it's true
```js
ws = new WebSocket("wss://echo.websocket.org", ['rpc.msgpack', 'rpc.json'])
```

Reply:
https://github.com/nodejs/undici/blob/c49058b956bda999bc26a1c43c66f1bede2ebc18/lib/web/websocket/connection.js#L111-L118

Reply:
Sorry, I didn't notice that this check happens earlier